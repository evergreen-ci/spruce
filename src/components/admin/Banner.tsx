import { Card, CardContent, CardHeader, TextField } from "@material-ui/core";
import * as React from "react";

interface Props {
  onBannerTextChange?: (text: string) => void;
  banner?: string;
}

export class BannerCard extends React.Component<Props> {
  public render() {
    return (
      <Card>
        <CardHeader title="Announcements" />
        <CardContent>
          <TextField
            id="bannerText"
            label="Banner Message"
            value={this.props.banner || ""}
            onChange={this.setBanner()}
          />
        </CardContent>
      </Card>
    );
  }

  private setBanner = () => (event: React.ChangeEvent<HTMLInputElement>) => {
    if (this.props.onBannerTextChange) {
      this.props.onBannerTextChange(event.currentTarget.value);
    }
  };
}
