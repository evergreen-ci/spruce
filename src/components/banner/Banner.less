.banner-container {
  padding-right: 100px;
}
.banner {
  padding-top: 12px;
  padding-bottom: 12px;
  padding-left: 40px;
  padding-right: 40px;
  background-color: #dff0d8 !important;
  border-color: #d6e9c6;
  color: rgba(0, 0, 0, 0.54);
}
.vertical-center {
  vertical-align: middle;
}
.close-banner {
  padding-top: 13px;
  position: absolute;
  right: 140px;
  color: rgba(0, 0, 0, 0.54);
}