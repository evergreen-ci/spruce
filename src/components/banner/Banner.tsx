import { Grid, Link, Paper } from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';
import * as React from 'react';
import '../../styles.css';

interface State {
  bannerIsHidden: boolean
}

class Props {
  public message: any
  public storageKey: string
  public onFinishStateUpdate: () => void
}

export class PatchContainer extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    let shouldHideBanner = false;
    if (localStorage.getItem(this.props.storageKey) !== null) {
      shouldHideBanner = true;
    }
    this.state = {
      bannerIsHidden: shouldHideBanner
    };
  }

  public render() {

    return (
      <div className="banner-container" hidden={this.state.bannerIsHidden}>
        <Paper className="banner">
          <Grid container={true}>
            <Grid item={true} xs={11}>
              <p className="vertical-center">
                {"Welcome to the new patches page! You can opt out of this page or report bugs "}
                <Link href={"https://evergreen.mongodb.com/settings"}>here</Link>.
              </p>
            </Grid>
            <Grid item={true} xs={1}>
              <CloseIcon onClick={this.hideBanner} className="close-banner" />
            </Grid>
          </Grid>
        </Paper>
      </div>
    );
  }

  private hideBanner = () => {
    localStorage.setItem(this.props.storageKey, "true");
    this.setState({
      bannerIsHidden: true
    }, this.props.onFinishStateUpdate);
  }
}

export default PatchContainer;