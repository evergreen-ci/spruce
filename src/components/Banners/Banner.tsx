import React from "react";
import styled from "@emotion/styled";
import Icon from "@leafygreen-ui/icon";
import { uiColors } from "@leafygreen-ui/palette";
import { Body } from "@leafygreen-ui/typography";
import { wordBreakCss } from "components/Typography";
import { BannerTypeKeys } from "types/banner";

const { green, red, yellow, blue, gray } = uiColors;

interface Props {
  id?: string;
  type: BannerTypeKeys;
  removeBanner?: (bannerId: string) => void;
}

export const Banner: React.FC<Props> = ({
  children,
  type,
  removeBanner,
  id,
}) => {
  const onClickX = () => removeBanner(id);
  return (
    <Wrapper type={type} data-cy="banner">
      <StyledBody type={type}>{children}</StyledBody>
      {(type === "success" || type === "info") && (
        <X onClick={onClickX}>
          <Icon glyph="X" fill={gray.base} />
        </X>
      )}
    </Wrapper>
  );
};

const mapTypeToBorderColor: { [key in BannerTypeKeys]: string } = {
  success: green.light2,
  error: red.light2,
  warning: yellow.light2,
  info: blue.light2,
};
const mapTypeToBackgroundColor: { [key in BannerTypeKeys]: string } = {
  success: green.light3,
  error: red.light3,
  warning: yellow.light3,
  info: blue.light3,
};
const mapTypeToTextColor: { [key in BannerTypeKeys]: string } = {
  success: green.dark2,
  error: red.dark2,
  warning: yellow.dark2,
  info: blue.dark2,
};

interface StyleProps {
  type: BannerTypeKeys;
}

const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 16px 16px;
  border: 1px solid;
  border-radius: 4px;
  border-color: ${({ type }: StyleProps) => mapTypeToBorderColor[type]};
  background-color: ${({ type }: StyleProps) => mapTypeToBackgroundColor[type]};
`;
const StyledBody = styled(Body)`
  color: ${({ type }: StyleProps) => mapTypeToTextColor[type]};
  ${wordBreakCss}
`;
const X = styled.div`
  margin-left: 16px;
  cursor: pointer;
`;
