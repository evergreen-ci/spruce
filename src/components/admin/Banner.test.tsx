import * as enzyme from "enzyme";
import * as React from "react";
import { BannerCard } from "./Banner";

describe("BannerCard", () => {
  const onChange = jest.fn((value: string) => expect(value).toBe("new banner"));
  const wrapper = enzyme.shallow(
    <BannerCard banner="hello world" onBannerTextChange={onChange} />
  );

  it("renders", () => {
    expect(wrapper.html()).toContain("hello world");
    expect(wrapper.html()).toContain("Banner Message");
  });

  it("typing in field calls onChange", () => {
    wrapper
      .find("#bannerText")
      .simulate("change", { currentTarget: { value: "new banner" } });
    expect(onChange).toHaveBeenCalled();
  });
});
