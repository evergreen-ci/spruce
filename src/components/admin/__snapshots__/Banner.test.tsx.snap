// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerCard matches snapshot 1`] = `ShallowWrapper {}`;
